src/main/java/io/snowdrop/reporting/weekreport/WeekReportEndpoint.java

```diff
@@ -0,0 +1,150 @@
+    Set<String> users;
+
+    @ConfigProperty(name = "github.reporting.target.organization")
+    String repoOrganization;
+
+    @ConfigProperty(name = "github.reporting.target.repository")
+    String repoName;
+
+    @Inject
+    GitHubClient client;
+
+    @Inject
+    GithubReportingService service;
+
+    private void populate(Date startTime, Date endTime) {
+        Set<PullRequest> pullRequests = service.getPullRequestCollector().getPullRequests().filter(p -> p.isActiveDuring(startTime, endTime))
+                .collect(Collectors.toSet());
+        Set<Issue> issues = service.getIssueCollector().getIssues().filter(i -> i.isActiveDuring(startTime, endTime)).collect(Collectors.toSet());
+
+        Set<RepositoryWork> workItems = service.getRepositoryCollector().getUsers().stream().flatMap(u -> {
+            Map<String, Set<Issue>> userIssues = issues.stream().filter(i -> u.equals(i.getAssignee()))
+                    .collect(Collectors.groupingBy(i -> i.getRepository(), Collectors.toSet()));
+            Map<String, Set<PullRequest>> userPullRequests = pullRequests.stream().filter(p -> u.equals(p.getCreator()))
+                    .collect(Collectors.groupingBy(p -> p.getRepository(), Collectors.toSet()));
+            return Stream.concat(userIssues.keySet().stream(), userPullRequests.keySet().stream())
+                    .distinct()
+                    .map(r -> new RepositoryWork(Github.user(r), Github.repo(r), u,
+                            userIssues.getOrDefault(r, Collections.emptySet()),
+                            userPullRequests.getOrDefault(r, Collections.emptySet())));
+        }).collect(Collectors.toSet());
+    }
+
+    @GET
+    @Path("/generate")
+    @Produces(MediaType.TEXT_PLAIN)
+    @Transactional
+    public String createReport(@QueryParam("startTime") String startTimeString, @QueryParam("endTime") String endTimeString) {
+        String mdText;
+        String weekNumber;
+        IssueService issueService = new IssueService(client);
+        try {
+            Date startTime = startTimeString != null ? DF.parse(startTimeString)
+                    : Date.from(service.getPullRequestCollector().getStartTime().toInstant());
+            Date endTime = endTimeString != null ? DF.parse(endTimeString)
+                    : Date.from(service.getPullRequestCollector().getEndTime().toInstant());
+            weekNumber = WEEK_YEAR_FORMAT.format(endTime);
+            LOGGER.debug("week number: " + weekNumber);
+            populate(startTime, endTime);
+            mdText = WeeklyDevelopmentReportImpl.build(startTime, endTime, users).buildWeeklyReport(Repository.findAll().list());
+            LOGGER.info("weekly_development: " + mdText);
+            updateGithubIssue(mdText, "weekly_development_" + weekNumber, "report", issueService, repoOrganization, repoName);
+            mdText = DevelopmentReportImpl.build(startTime, endTime, users).buildWeeklyReport(Repository.findAll().list());
```

Comment:
Maybe the name is confusing. so what is then the job done by `WeeklyDevelopmentReportImpl` vs `DevelopmentReportIml` ?

Reply:
Ok, it sounds good to me.

Reply:
Yes, I understand the name is confusing but I couldn't do better :-(.

*WeeklyDevelopmentReportImpl* generates the weekly development report from the Issue in our team repository.

The *DevelopmentReportIml* generates the report with all the Issue and PR from all the github repositories we participate in.

Reply:
Makes sense